src/OrganisationUnitTree/Label.js

```diff
@@ -0,0 +1,243 @@
+    <SingleSelectionLabel
+        checked={false}
+        loading={loading}
+        label={label}
+        onChange={onToggleOpen}
+    />
+)
+
+DisabledSelectionLabel.propTypes = {
+    label: propTypes.string.isRequired,
+    onToggleOpen: propTypes.func.isRequired,
+    loading: propTypes.bool,
+}
+
+/**
+ * @param {Object} props
+ * @param {string} props.label
+ * @param {bool} [props.checked]
+ * @param {bool} [props.loading]
+ * @param {Function} [props.onChange]
+ * @returns {React.Component}
+ */
+const SingleSelectionLabel = ({ checked, label, onChange, loading }) => (
+    <span onClick={onChange} className={cx({ checked, loading })}>
+        {label}
+
+        <style jsx>{`
+            span {
+                display: inline-block;
+                background: transparent;
+                border-radius: 3px;
+                padding: 0 5px;
+                line-height: 26px;
+                white-space: nowrap;
+                cursor: pointer;
+                user-select: none;
+            }
+
+            .checked {
+                background: ${colors.teal700};
+                color: white;
+            }
+
+            .loading {
+                cursor: auto;
+            }
+        `}</style>
+    </span>
+)
+
+SingleSelectionLabel.propTypes = {
+    label: propTypes.string.isRequired,
+    checked: propTypes.bool,
+    loading: propTypes.bool,
+    onChange: propTypes.func,
+}
+
+/**
+ * @param {Object} props
+ * @param {string} props.id
+ * @param {Object} props.error
+ * @param {string} props.error.message
+ * @param {bool} [props.singleSelectionOnly]
+ * @returns {React.Component}
+ */
+const ErrorMessage = ({ id, error, singleSelectionOnly }) => (
+    <span className={cx({ 'with-checkbox': !singleSelectionOnly })}>
+        {`
+            An error occured loading organisation unit with id "${id}".
+            Error: ${error.message}
+        `}
+
+        <style jsx>{`
+            span {
+                display: inline-block;
+                padding-left: 5px;
+                color: ${colors.red500};
+            }
+
+            .with-checkbox {
+                padding-left: 32px;
+            }
+        `}</style>
+    </span>
+)
+
+ErrorMessage.propTypes = {
+    id: propTypes.string.isRequired,
+    error: propTypes.shape({ message: propTypes.string }).isRequired,
+    singleSelectionOnly: propTypes.bool,
+}
+
+/**
+ * @module
+ * @param {Label.PropTypes} props
+ * @returns {React.Component}
+ */
+const Label = ({
+    id,
+    path,
+    open,
+    loading,
+    error,
+    checked,
+    onChange,
+    hasChildren,
+    highlighted,
+    displayName,
+    onToggleOpen,
+    disableSelection,
+    singleSelectionOnly,
+    hasSelectedDescendants,
+}) => {
+    const label = !displayName && loading ? 'Loading...' : displayName
+    const onClick = event => {
+        const newChecked = event.target.checked
+        onChange({ id, path, checked: newChecked }, event)
+    }
```

Comment:
After having read [the article](https://cdb.reacttraining.com/react-inline-functions-and-performance-bdff784f5578) @ismay shared in a PR, I think we shouldn't worry at all until we notice that there's a problem. We could do a performance measure with and without memoizing stuff.

Right now I'm inclined to remove all the memoization stuff, because we don't know if it's actually hurting in the long run.